### PR TITLE
hal: Fix buffer overflow error in XFindFirstFile/XFindNextFile

### PR DESCRIPTION
**Pull request by thrimbor**
**/edit: Ignore this for now, I plan to submit a PR which replaces all this code with a WinAPI-style implementation.**

This is the issue @dracc encountered and described in Discord.

The issue was that these functions directly used `FILE_DIRECTORY_INFORMATION` and passed its size to `NtQueryDirectoryFile`, which is problematic because `FILE_DIRECTORY_INFORMATION` contains a field of variable length called `FileName` at its end.
It seems that OpenXDK used to hack around this by giving this field a fixed length of 0x100.

I instead solved this by reserving a buffer with a length of `sizeof(FILE_DIRECTORY_INFORMATION) + 1024` and casting it to a `PFILE_DIRECTORY_INFORMATION`. I checked afterwards how ReactOS does this, and they utilize a similar approach but with a larger buffer. Considering the fact that this buffer is placed on the stack, I decided to go with the smaller buffer size.

While being a bit ugly, this solution allows us to keep our Kernel-header close to MS documentation / ReactOS.

### CODE REVIEW

File: lib/hal/fileio.c [hunk 1]
```diff
@@ -6,6 +6,9 @@
 #include <xboxrt/debug.h>
 #include <stdbool.h>
 
+// Defines the allowed size for a FILE_DIRECTORY_INFORMATION entry
+#define FIND_DATA_SIZE (sizeof(FILE_DIRECTORY_INFORMATION) + 1024)
```
Comment (JayFoxRox) [inline lib/hal/fileio.c]: Do we have a `MAX_PATH`? Should we have one?

File: lib/hal/fileio.c [hunk 2]
```diff
@@ -802,34 +806,35 @@ int XFindNextFile(
 #endif
 
 	IO_STATUS_BLOCK IoStatusBlock;
-	FILE_DIRECTORY_INFORMATION FileInformation;
+	BYTE FileInfoRaw[FIND_DATA_SIZE];
+	PFILE_DIRECTORY_INFORMATION FileInformation = (PFILE_DIRECTORY_INFORMATION)FileInfoRaw;
```
Comment (JayFoxRox) [inline lib/hal/fileio.c]: nit: The standard way of handling this would be a union I think.

**Comment by JayFoxRox** [comment]
This has been sitting idle for a while, and it was supposed to be replaced by a WinAPI option.

Should this PR be closed / rejected; or should it be merged as a temporary measure?

From what I can tell, it's "better than master". I did not test it yet, but did review a while ago.

**Comment by mborgerson** [comment]
Closing for now as it seems like this PR is no longer necessary with #112

**Pull request closed without merging**